src/flb_mp.c

```diff
@@ -382,7 +413,13 @@ struct flb_mp_accessor *flb_mp_accessor_create(struct mk_list *slist_patterns)
             flb_mp_accessor_destroy(mpa);
             return NULL;
         }
-        mk_list_add(&ra->_head, &mpa->ra_list);
+
+        /* first time */
+        if (mk_list_size(&mpa->ra_list) == 0) {
```

Comment:
I think you could skip this block if becaise the `mk_list_foreach` in `insert_by_subkey_num` would not satisfy the condition in the for loop when the list is empty so it would skip straight to the `mk_list_add` fallback right?

Reply:
Thank you for reviewing. You are right.

I sent a following commit.
https://github.com/fluent/fluent-bit/pull/5590/commits/b98c0633054cb8381de007b4acdcbcfd0c4e3e28